share/eds/byzantine/bad_encoding.go

```diff
@@ -138,7 +138,7 @@ func (p *BadEncodingProof) Validate(header *header.ExtendedHeader) error {
 		if share == nil {
 			continue
 		}
-		shares[index] = share.Share
```

Comment:
This seems suspicious. @evan-forbes, do you know why we now need to cut the namespace?

Reply:
yes what @walldiss said. @Wondertan, perhaps it would be better to implement this change elsewhere? https://github.com/celestiaorg/celestia-node/blob/a95b114cf0353a4760314bcd198e8cd9c831f83b/share/eds/byzantine/share_proof.go#L91

unlike the old codec, the new one requires chunks of data be length 256 or 512. I'm not the most familiar with this codebase, so I'm far from certain, but it might have been a bug if we're BEFPs using the extra namespace added by the nmt wrapper to the erasure code.

Reply:
Wait, so now there are twice prepended namespace instead of 1(as we received before), right? Or I missed something?

Reply:
There must be a bug somewhere I feel

Reply:
Well, i was wrong with 2 namespaces assumption.  The thing is the share that is passed to validator is original share (1 namespace + 512 bytes). Later, inside the validation function namespace will be cut .
 But the `shares` slice that is filled in line 141 contains only Share data (512 bytes), later elements of this slice passed to Decoder. 
 It used to work before because Decodor didn't check `Share` length and consumed it with prepended namespace without any sign of error.

Reply:
found the bug if my logic is correct, pls see https://github.com/celestiaorg/celestia-node/issues/1355 and https://github.com/celestiaorg/celestia-node/pull/1354

Reply:
the test in #1354 passes on this branch but fails on main